src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -698,10 +698,14 @@ trait TypeDiagnostics {
           val opc = new overridingPairs.Cursor(classOf(m))
           opc.iterator.exists(pair => pair.low == m)
         }
+        import PartialFunction._
         def isConvention(p: Symbol): Boolean = {
-          (p.name.decoded == "args" && p.owner.isMethod && p.owner.name.decoded == "main") ||
-            (p.tpe =:= typeOf[scala.Predef.DummyImplicit])
-        }
+          val ds = currentRun.runDefinitions
+          import ds._ ; (
+            p.name.decoded == "args" && p.owner.isMethod && p.owner.name.decoded == "main"
+          ||
+            p.isImplicit && cond(p.tpe.typeSymbol) { case Predef_=:= | Predef_<:< | Predef_Dummy => true }
+        )}
```

Comment:
I hope someone appreciates the fancy bracketing, or suggests something prettier.

Reply:
Excepting that it appears to slightly exceed 80 characters, it doesn't look all that bad. It's kind of a pity that `import currentRun.runDefinitions._` won't ever work. I think I've seen it called `rd` elsewhere, but I don't think that matters.

Reply:
`currentRun.runDefinitions.{ someday }`